### PR TITLE
CSCwi89338: Applying latest rpm build tool changes

### PR DESCRIPTION
**Pull request by nburnwalcisco**
Enhanced published rpm build tool by merging internal build rpm tool changes

### CODE REVIEW

File: docker/Centos.Dockerfile [hunk 1]
```diff
@@ -1,6 +1,11 @@
-FROM centos:7
+FROM centos:8
 
+RUN cd /etc/yum.repos.d/
+RUN sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-*
+RUN sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-*
 RUN yum install -y pinentry rpm-sign rpm-build
 
 COPY build_rpm.sh /usr/sbin/
+# FROM akshshar/xr-wrl7
```
Comment (arshanh) [inline docker/Centos.Dockerfile:9]: Remove these commented lines

File: docker/Centos.Dockerfile [hunk 2]
```diff
@@ -1,6 +1,11 @@
-FROM centos:7
+FROM centos:8
 
+RUN cd /etc/yum.repos.d/
+RUN sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-*
```
Comment (arshanh) [inline docker/Centos.Dockerfile:4]: Can leave a comment explaining the `sed`

File: examples/alpine/build.yaml [hunk 3]
```diff
@@ -1,28 +1,59 @@
+#Refer README.md for more info
 packages:
-- name: "alpine"
-  release: "ThinXR_7.3.15"
-  version: "0.1.0"
+- name: "alpine" #Can be changed
+  release: "ThinXR_7.3.15" #Do not change
+  target-release: "ThinXR_7.3.15" #Can be changed
```
Comment (arshanh) [inline examples/alpine/build.yaml:5]: What purpose does target-release serve?

File: docker/Centos.Dockerfile (hunk 1 above)
Comment (nburnwalcisco) [reply docker/Centos.Dockerfile]: Done

File: docker/Centos.Dockerfile (hunk 2 above)
Comment (nburnwalcisco) [reply docker/Centos.Dockerfile:5]: Done

File: examples/alpine/build.yaml (hunk 3 above)
Comment (nburnwalcisco) [reply examples/alpine/build.yaml:5]: target-release, as the comment says, "can be modified" and corresponds to the release for which rpm is being built. Also this release is appended to the rpm name.

File: docker/Centos.Dockerfile (hunk 2 above)
Comment (arshanh) [reply docker/Centos.Dockerfile:5]: but why do we need to comment all the occurrences of the mirrorlist? :)

**Review by arshanh — APPROVED** [review]

File: docker/Centos.Dockerfile (hunk 2 above)
Comment (nburnwalcisco) [reply docker/Centos.Dockerfile:5]: When adding support for sandbox rpm build, we were facing below error when running dnf commands,

dnf.exceptions.RepoError: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist
2022-01-31T15:28:05+0000 CRITICAL Error: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist

We followed https://forketyfork.medium.com/centos-8-no-urls-in-mirrorlist-error-3f87c3466faa and made the suggested changes to comment mirrorlist and change baseurl,

sudo sed -i -e "s|mirrorlist=|#mirrorlist=|g" /etc/yum.repos.d/CentOS-*
sudo sed -i -e "s|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g" /etc/yum.repos.d/CentOS-*

**Review by sumitk31 — APPROVED** [review]

**Pull request merged**